### PR TITLE
feat(option): add internaloption to force use of certain credential

### PR DESCRIPTION
**Pull request by BrennaEpp**
A new internaloption added that forces the client to use a certain credential. This ensures that if a user directly passes in their own credentials, we can capture and use these credentials.

This is part of the effort to improve Signed URLs in the Go Storage Client.

### CODE REVIEW

File: option/internaloption/internaloption.go [hunk 1]
```diff
@@ -106,3 +107,15 @@ type enableJwtWithScope bool
 func (w enableJwtWithScope) Apply(o *internal.DialSettings) {
 	o.EnableJwtWithScope = bool(w)
 }
+
+// WithCredentials returns a client option to specify authenticates API calls.
```
Comment (tritone) [inline option/internaloption/internaloption.go:111]: Seems like this sentence is a fragment-- should it be something like "a client option to specify credentials which will be used to authenticate API calls"?

**Review by tritone — APPROVED** [review]
Generally seems fine to me-- @codyoss @broady thoughts?

File: option/internaloption/internaloption.go (hunk 1 above)
Comment (BrennaEpp) [reply option/internaloption/internaloption.go]: good catch!

**Comment by codyoss** [comment]
LGTM, but I worked on this code so I will let @broady review.

**Review by broady — APPROVED** [review]

**Pull request merged**